compiler/src/dotty/tools/dotc/transform/LetOverApply.scala

```diff
@@ -0,0 +1,31 @@
+package dotty.tools
+package dotc
+package transform
+
+import core._
+import Contexts.Context, Symbols._, Decorators._
+import MegaPhase._
+import ast.Trees._
+
+/** Rewrite `{ stats; expr}.f(args) }` to `{ stats; expr.f(args) }` before
+ *  proceeding, but leave closures alone. This is necessary to be able to
+ *  collapse applies of IFTs (this is done in Erasure).
```

Comment:
If I understand this correctly, it also rewrites `{ stats; expr }(args) }` to `{ stats; expr(args) }`:
```suggestion
/** Rewrite `{ stats; expr}.f(args) }` to `{ stats; expr.f(args) }` and
 *  `{ stats; expr }(args) }` to `{ stats; expr(args) }` before proceeding,
 *  but leave closures alone. This is necessary to be able to
 *  collapse applies of IFTs (this is done in Erasure).
```